DotBot.py

```diff
@@ -58,7 +60,8 @@ def receive(self, frame):
             return
         self.lastMovementSeqNum   = myMovement['movementSeqNum']
 
-        # if I get here I have receive a NEW movement
+        # if I get here I have received a NEW movement
+        self.newMovementReceived = True
```

Comment:
I don't understad why you need the `self.newMovementReceived` attribute. If you get to this line of code, simply cancel all future retransmission events.